src/egit-ignore.c

```diff
@@ -0,0 +1,59 @@
+#include "git2.h"
+
+#include "egit.h"
+#include "interface.h"
+
+EGIT_DOC(add_rule, "REPO RULES",
+         "Add ignore rules listed in RULES to libgit2s internal list of ignore \
+         rules for REPO. These rules are merged with any rules listed in a \
+         .gitignore file in the repo. Rules are separated by newlines.");
+emacs_value egit_add_rule(emacs_env *env, emacs_value _repo, emacs_value _rules)
+{
+    EGIT_ASSERT_REPOSITORY(_repo);
+    EGIT_ASSERT_STRING(_rules);
+
+    git_repository *repo = EGIT_EXTRACT(_repo);
+    int retval;
+    {
+        char *rules = EGIT_EXTRACT_STRING(_rules);
+        retval = git_ignore_add_rule(repo, rules);
+        free(rules);
+    }
+    EGIT_CHECK_ERROR(retval);
+
+    return env->make_integer(env, retval);
```

Comment:
I usually just returm `em_nil` in cases like this to be honest. Doesn't really matter either way.